src/Context/CachedContextFactory.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Roave\BetterReflection\Context;
+
+use Roave\BetterReflection\Context\ContextFactory;
+
+class CachedContextFactory implements ContextFactory
+{
+    private $contexts = [];
```

Comment:
guess that makes sense. In that case would we instantiate the `Reflector` with a single cache implementation? Does it make sense to implement it in this PR or a follow-up?